scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -419,40 +462,78 @@ private void receiveNextDatagramFromNetwork() throws IOException {
 		LOGGER.log(Level.FINER, "Received {0} DTLS records using a {1} byte datagram buffer",
 				new Object[]{records.size(), inboundDatagramBufferSize});
 
-		for (Record record : records) {
+		for (final Record record : records) {
 			try {
-				LOGGER.log(Level.FINEST, "Received DTLS record of type [{0}]", record.getType());
 
 				switch(record.getType()) {
+				case HANDSHAKE:
 				case APPLICATION_DATA:
-					processApplicationDataRecord(record);
-					break;
 				case ALERT:
-					processAlertRecord(record);
-					break;
 				case CHANGE_CIPHER_SPEC:
-					processChangeCipherSpecRecord(record);
-					break;
-				case HANDSHAKE:
-					processHandshakeRecord(record);
+					executor.execute(new StripedRunnable() {
+
+						@Override
+						public Object getStripe() {
+							return record.getPeerAddress();
```

Comment:
That's true. Very good point. At least we were able to detect that an abbreviated handshake was already going on ...
Since we're synchronizing access to the `ConnectionStore` we should be able to still detect this situation and e.g. reject the concurrent resumption request.